spirit/category/admin/forms.py

```diff
@@ -36,3 +37,11 @@ def clean_parent(self):
                                               "can not have a parent since it has childrens"))
 
         return parent
+
+    def clean_colour(self):
+        colour = self.cleaned_data["colour"]
+
+        if colour != '' and not re.match(r'^#[a-f0-9]{3}([a-f0-9]{3}){0,1}$', colour):
+            raise forms.ValidationError(_("The colour you are updating "
+                                          "is not a valid hex colour string"))
+        return colour
```

Comment:
Remove the *updating* part of the message, since this is also shown when creating a category. Maybe "*The input is not a valid hex color.*" I dunno (don't like the input word, but don't wanna repeat the "color" word).